runtime/resolver.go

```diff
@@ -139,28 +148,44 @@ func (r *Runtime) Resolve(ctx context.Context, opts *ResolveOptions) ([]byte, er
 
 	// Try to get from cache
 	if val, ok := r.queryCache.cache.Get(key); ok {
-		return val.([]byte), nil
+		res, ok := val.(*CachedResolverResult)
+		if ok {
+			return res, nil
+		}
+		// rerun the query if cache cannot be converted to *CachedResolverResult
```

Comment:
Having an unconditional cast is actually preferable. This would be a severe developer error and it should rightfully panic in this case.